app/models/miq_report/generator.rb

```diff
@@ -911,6 +911,6 @@ def interval
   end
 
   def custom_results_method
-    @custom_results_method ||= db_options && db_options[:rpt_type] ? "build_results_for_report_#{db_options[:rpt_type]}" : nil
+    @custom_results_method ||= db_options.try(:[], :rpt_type) && "build_results_for_report_#{db_options[:rpt_type]}"
```

Comment:
This one feels weird to me if only because its doing a double lookup (but it did do that before, so it's probably ok)

Reply:
so many things wrong with this. redoing